code/modules/vending/mail.dm

```diff
@@ -0,0 +1,308 @@
+	currentstate = STATE_SORTING
+	update_appearance()
+	if (!sort_delay())
+		return
+	for (var/obj/item/mail/some_mail in mail_list)
+		if (!some_mail.recipient_ref)
+			unable_to_sort ++
+			continue
+		var/datum/mind/some_recipient = some_mail.recipient_ref.resolve()
+		if (some_recipient)
+			var/datum/job/recipient_job = some_recipient.assigned_role
+			var/datum/job_department/primary_department = recipient_job.departments_list?[1]
+			var/datum/job_department/main_department = primary_department.department_name
+			if (main_department == sorting_dept)
+				sorted_mail.Add(some_mail)
+				sorted ++
+		else
+			unable_to_sort ++
+	if (length(sorted_mail) == 0)
+		currentstate = STATE_NO
+		update_appearance()
+		playsound(src, 'sound/machines/buzz/buzz-sigh.ogg', 20, TRUE)
+		say("No mail for the following department: [sorting_dept].")
+	else
+		currentstate = STATE_YES
+		update_appearance()
+		say("[sorted] envelope\s sorted successfully.")
+		playsound(src, 'sound/machines/ping.ogg', 20, TRUE)
+		to_chat(usr, span_notice("[src] ejects [length(sorted_mail)] envelope\s."))
+		var/turf/unload_turf = get_unload_turf()
+		for (var/obj/item/mail/mail_in_list in sorted_mail)
+			mail_in_list.forceMove(unload_turf)
+			sorted_mail -= mail_in_list
+			mail_list -= mail_in_list
+	sleep(10)
+	if (unable_to_sort > 0)
+		playsound(src, 'sound/machines/buzz/buzz-sigh.ogg', 20, TRUE)
+		say("Couldn't sort [unable_to_sort] envelope\s.")
+	else
+		playsound(src, 'sound/machines/ping.ogg', 20, TRUE)
+		say("[total_to_sort] envelope\s processed.")
+	sleep(10)
+	currentstate = STATE_IDLE
+	update_appearance()
+
+/obj/machinery/mailsorter/attackby(obj/item/thingy, mob/user, params)
+	if (istype(thingy, /obj/item/storage/bag/mail))
+		if (length(thingy.contents) < 1)
+			to_chat(user, span_warning("The [thingy] is empty!"))
+			return
+		var/loaded = 0
+		for (var/obj/item/mail in thingy.contents)
+			if (!(mail.item_flags & ABSTRACT) && \
+				!(mail.flags_1 & HOLOGRAM_1) && \
+				accept_check(mail) \
+			)
+				if (length(mail_list) + 1 > mail_capacity)
+					to_chat(user, span_warning("There is no space for more mail in [src]!"))
+					return FALSE
+				else if (load(mail, usr))
+					loaded++
+					mail_list += mail
+		if(loaded)
+			user.visible_message(span_notice("[user] loads \the [src] with \the [thingy]."), \
+			span_notice("You load \the [src] with \the [thingy]."))
+			if(length(thingy.contents))
+				to_chat(user, span_warning("Some items are refused."))
+			return TRUE
+		else
+			to_chat(user, span_warning("There is nothing in \the [thingy] to put in the [src]!"))
+			return FALSE
+	else if (istype(thingy, /obj/item/mail))
+		if (length(mail_list) + 1 > mail_capacity)
+			to_chat(user, span_warning("There is no space for more mail in [src]!"))
+		else
+			thingy.forceMove(src)
+			mail_list += thingy
+			to_chat(user, span_notice("The [src] whizzles as it accepts the [thingy]."))
+	. = ..()
+
+/obj/machinery/mailsorter/proc/pick_mail(usr)
```

Comment:
Same here about replacing with ``mob/user``

Reply:
Uh, what for, if I'm already passing user here?

Reply:
Like I said in the other review, rename the arg to mob/user instead of just usr, and use user instead of usr. "Usr" is a byond thing and you shouldn't mix the two up lest you create beasts that will destroy our beloved codebase.